iceoryx_hoofs/source/posix_wrapper/signal_watcher.cpp

```diff
@@ -29,26 +29,18 @@ void internalSignalHandler(int) noexcept
     for (uint64_t remainingNumberOfWaiters = instance.m_numberOfWaiters.load(); remainingNumberOfWaiters > 0;
          --remainingNumberOfWaiters)
     {
-        instance.m_semaphore.post().or_else([](auto) {
-            constexpr const char MSG[] = "Unable to increment semaphore in signal handler";
-            auto result = write(STDERR_FILENO, MSG, sizeof(MSG));
-            IOX_DISCARD_RESULT(result);
-            std::abort();
-        });
+        instance.m_semaphore->post().expect("Unable to increment semaphore in signal handler");
```

Comment:
Hehe, you cannot call `expect` in this function since `expect` might call the error handler and the error handler is not signal safe